mpqp/core/instruction/gates/native_gates.py

```diff
@@ -902,6 +911,9 @@ def __init__(self, control: int, target: int):
     def to_matrix(self) -> Matrix:
         return np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])
 
+    def __repr__(self) -> str:
+        return f"{type(self).__name__}({self.controls[0]}, {self.targets[0]})"
```

Comment:
Don't you think that we can factorize that code for all controlled gates and have an automated code that handles single and multi-control (like for TOF) for the __repr__ ?
This would avoid replicating the same code for all controlled gates. The only particular case would be CRk, but it is already there so it can overcharge the one we would write for ControlledGate.

Reply:
I actually don't think it's the case, because a controlled gate could have multiple control qubits. If we really wanted to factorize the behavior I guess we could have a generic implementation for `ControlledGate`s and add a class like `UniControlledGate` for gates with only one control qubit, but at this point I'm not sure enough gates are concerned by this to make it worth having this change. I'll let you convince me otherwise

Reply:
for demonstration purpose I did what I mentioned in the branch feat-gates-repr-facto-experience (commit 0e5f0f8), I'll let you be judge if you want it or not

Reply:
@Henri-ColibrITD @MoHermes I proposed another version that does not require the definition of another class but checks directly the `targets` and `controls` attributes to format correctly the `__repr__method ( see branch https://github.com/ColibrITD-SAS/mpqp/tree/feat-gates-repr-facto-2)